src/App.jsx

```diff
@@ -14,70 +16,73 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
+export const App = () => {
+  const [good, setGood] = useState('Jam');
+  const noGoodSelectedBlock = (
     <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+  );
 
+  const goodSelectedBlock = (
     <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+      {good}
+      {' is selected'}
 
       <button
         data-cy="ClearButton"
         type="button"
         className="delete ml-3"
+        onClick={() => setGood('')}
       />
     </h1>
+  );
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  return (
+    <>
+      <main className="section container">
+        {good === ''
+          ? noGoodSelectedBlock
+          : goodSelectedBlock
+        }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map((goodElement, index) => {
+              const isSelected = good === goodElement;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              return (
+                <tr
+                  data-cy="Good"
+                  className={cn({
+                    'has-background-success-light': isSelected,
+                  })}
+                  key={goodElement}
+                >
+                  <td>
+                    <button
+                      data-cy={
+                        isSelected
+                          ? 'RemoveButton'
+                          : 'AddButton'
+                      }
+                      type="button"
+                      className={cn('button', {
+                        'is-info': isSelected,
+                      })}
+                      onClick={() => setGood(isSelected ? '' : goodElement)}
```

Comment:
```suggestion
                      onClick={() => selectGood(isSelected, goodElement)}
```

I'd recommend creating a separate function for this one as well.

![image](https://github.com/mate-academy/react_goods-selector-js/assets/102293862/956bc8b3-4e15-4a0e-9552-e4dc903e2d85)

This way the code will be more straightforward and easier to support.